src/Okta.Idx.Sdk/IIdxClient.cs

```diff
@@ -150,5 +150,21 @@ public interface IIdxClient : IOktaClient
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>The authentication response.</returns>
         Task<AuthenticationResponse> SkipAuthenticatorSelectionAsync(IIdxContext idxContext, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Gets available identity providers.
+        /// </summary>
+        /// <param name="state">The state handle to use or null to autogenerate.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The identity providers response.</returns>
+        Task<IdentityProvidersResponse> GetIdentityProvidersAsync(string state = null, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Gets available identity providers.
+        /// </summary>
+        /// <param name="idxContext">The IDX context.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The identity providers response.</returns>
+        Task<IdentityProvidersResponse> GetIdentityProvidersAsync(IIdxContext idxContext, CancellationToken cancellationToken = default);
```

Comment:
Question for my own knowledge, in what cases we use this one or the other overload with `state`?

Reply:
The idea is to call this overload if you already have a context to avoid an extra call to `Interact` if possible.